wagtail_localize/models.py

```diff
@@ -876,6 +879,75 @@ def schema_out_of_date(self):
         )
         return self.schema_version != current_schema_version
 
+    def sync_view_restrictions(self, original, translation_page):
+        """
+        Synchronizes view restriction object for the translated page
+
+        Args:
+            original (Page|Snippet): The original instance.
+            translation_page (Page|Snippet): The translated instance.
+        """
+        if not isinstance(original, Page) or not isinstance(translation_page, Page):
+            return
+
+        if original.view_restrictions.exists():
+            original_restriction = original.view_restrictions.first()
+            if not translation_page.view_restrictions.exists():
+                view_restriction, child_object_map = _copy(
+                    original_restriction,
+                    exclude_fields=["id"],
+                    update_attrs={"page": translation_page},
+                )
+                view_restriction.save()
+            else:
+                # if both exist, sync them
+                translation_restriction = translation_page.view_restrictions.first()
+                should_save = False
+                if (
+                    translation_restriction.restriction_type
+                    != original_restriction.restriction_type
+                ):
+                    translation_restriction.restriction_type = (
+                        original_restriction.restriction_type
+                    )
+                    should_save = True
+                if translation_restriction.password != original_restriction.password:
+                    translation_restriction.password = original_restriction.password
+                    should_save = True
+                if list(
+                    original_restriction.groups.values_list("pk", flat=True)
+                ) != list(translation_restriction.groups.values_list("pk", flat=True)):
+                    translation_restriction.groups.set(
+                        original_restriction.groups.all()
+                    )
```

Comment:
Are wemissing

`should_save = True`

in this last if block?

Reply:
Also, just for my own understanding, is the reason why we're doing this

```py
should_save = False
if this != that:
    this = that
    should_save = True

if should_save:
    this.save()
```

pattern rather than just

```py
this = that
this.save()
```

for performance reasons? (i.e: we're trying to minimise round trips to the DB)

Reply:
it certainly is missing `should_save = True`.

The reason for this is that 2 or more of the conditions can be true.

e.g. when changing form the 'login' restriction to 'password', both the `restriction_type` and `password` attributes will have changed.

so we want a single save

Reply:
on missing `should_save`, we are not - https://github.com/wagtail/wagtail-localize/pull/496/commits/0741ae45233ed06f11a65d35c6995d31e6ff75e2#diff-a9c40e8134c27fd95a23649e865005e8a957dc24db0d423122655a565221d614R555-R565 - as groups are M2M